src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/__init__.py

```diff
@@ -4,46 +4,76 @@
-        self.extra('parent', options_list='--resource-parent', arg_group=arg_group, help="Target resource parent path, if applicable.")
-        self.extra('resource_type', options_list='--resource-type', arg_group=arg_group, help="Target resource type. Can also accept namespace/type format (Ex: 'Microsoft.Compute/virtualMachines)')")
-        self.extra('resource_group_name', options_list=('--resource-group', '-g'), arg_group='Target Resource')
+        from .validators import get_target_resource_validator
 
-    def resource_parameter(self, dest, arg_group=None, required=True):
-        """ Helper method to add the extra parameters needed to support specifying name or ID for target resources. """
-        from azure.cli.command_modules.monitor.validators import get_target_resource_validator
-        self.argument(dest, options_list=['--{}'.format(dest)], arg_group=arg_group, required=required, validator=get_target_resource_validator(dest, required, preserve_resource_group_parameter=True), help="Name or ID of the target resource.")
-        self.extra('namespace', options_list=['--{}-namespace'.format(dest)], arg_group=arg_group, help="Target resource provider namespace.")
-        self.extra('parent', options_list=['--{}-parent'.format(dest)], arg_group=arg_group, help="Target resource parent path, if applicable.")
-        self.extra('resource_type', options_list=['--{}-type'.format(dest)], arg_group=arg_group, help="Target resource type. Can also accept namespace/type format (Ex: 'Microsoft.Compute/virtualMachines)')")
+        self.argument(dest, options_list='--resource', arg_group=arg_group, required=required,
+                      validator=get_target_resource_validator(dest, required),
+                      help="Name or ID of the target resource.")
+        self.extra('namespace', options_list='--resource-namespace', arg_group=arg_group,
+                   help="Target resource provider namespace.")
+        self.extra('parent', options_list='--resource-parent', arg_group=arg_group,
+                   help="Target resource parent path, if applicable.")
+        self.extra('resource_type', options_list='--resource-type', arg_group=arg_group,
+                   help="Target resource type. Can also accept namespace/type format "
+                        "(Ex: 'Microsoft.Compute/virtualMachines)')")
+        self.extra('resource_group_name', options_list=('--resource-group', '-g'), arg_group='Target Resource')
 
 
 class MonitorCommandsLoader(AzCommandsLoader):
-
     def __init__(self, cli_ctx=None):
         super(MonitorCommandsLoader, self).__init__(cli_ctx=cli_ctx,
-                                                    min_profile='2017-03-10-profile',
                                                     argument_context_cls=MonitorArgumentContext)
 
+    def monitor_command_group(self, name, client_factory, sdk_module=None, sdk_class=None, custom_module=None,
+                              **kwargs):
```

Comment:
There's no reason to create a special command group command here. It's just additional complexity and clashes against all the other modules that don't do this.